src/prisma/schema.prisma

```diff
@@ -0,0 +1,93 @@
+// This is your Prisma schema file,
+// learn more about it in the docs: https://pris.ly/d/prisma-schema
+
+generator client {
+  provider = "prisma-client-js"
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+
+model User {
+  id                     String         @id @default(cuid())
+  email                  String?        @unique
+  address                String         @unique
+  listings               Listing[]
+  auctions               Auction[]
+  initiatedListingOffers ListingOffer[] @relation("initiatedListingOffers")
+  receievedListingOffers ListingOffer[] @relation("receivedListingOffers")
+  initiatedAuctionBids   AuctionBid[]   @relation("initiatedAuctionBids")
+  receivedAuctionBids    AuctionBid[]   @relation("receivedAuctionBids")
+}
+
+model Listing {
+  id          String         @id @default(cuid())
+  creatorAddr String
+  tokenId     String         @unique
+  price       Decimal
+  createdAt   DateTime       @default(now())
+  updatedAt   DateTime       @updatedAt()
+  offers      ListingOffer[]
+  nft         NFT            @relation(fields: [tokenId], references: [tokenId], onDelete: Cascade)
+  creator     User           @relation(fields: [creatorAddr], references: [address], onDelete: Cascade)
+}
+
+model ListingOffer {
+  id          String   @id @default(cuid())
+  offerorAddr String
+  offereeAddr String
```

Comment:
1. Is there really a need to have `offererAddr` and `offerorAddr` since the listing contains a reference to the creator -- the user that made the listing.

2. During the design, you mentioned that users can make offers without the owner listing the NFT, that does not seem to be handled here.

**Note: Also, offeror and offerer are synonyms.**

Reply:
1. Including offerorAddr and offereeAddr in the ListingOffer model can significantly improve performance. This approach avoids the need for extra queries or database joins when accessing the offeror or offeree. Given that reading data is a common operation, optimizing for read performance can greatly enhance the overall efficiency of our system.
2. I'd look into that

Also I wrote offeror and offeree and not offerer and offeror